uftrace.c

```diff
@@ -335,6 +340,21 @@ static bool has_time_unit(const char *str)
 		return false;
 }
 
+static void check_shmem_freespace()
+{
+	struct statvfs stat;
+	long space;
+
+	if (statvfs(SHMEM_PATH, &stat) != 0) {
+		pr_warn("Cannot access shared memory.\n");
+		return;
+	}
+
+	space = stat.f_bsize * stat.f_bavail;
+	if (space < MIN_SHMEM_SPACE)
+		pr_warn("Freespace of shared memory is too small.\n");
```

Comment:
What do you want to do here?  Show warning and proceed?

Reply:
I was just asking you intended to proceed in this case rather than exit.  I'm ok with it and there's a chance that kernel can reclaim pages to make some available shmem.